src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs

```diff
@@ -258,14 +258,16 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, bool doRe
                     proxyUri = _proxy.GetProxy(request.RequestUri);
                 }
             }
-            catch (Exception)
+            catch (Exception ex)
             {
                 // Eat any exception from the IWebProxy and just treat it as no proxy.
                 // This matches the behavior of other handlers.
+                if (NetEventSource.IsEnabled) NetEventSource.Error(this, ex);
             }
 
             if (proxyUri != null && proxyUri.Scheme != UriScheme.Http)
             {
+                if (NetEventSource.IsEnabled) NetEventSource.Error(this, SR.net_http_invalid_proxy_scheme);
                 throw new NotSupportedException(SR.net_http_invalid_proxy_scheme);
```

Comment:
Don't exceptions like this show up in traces already?

Reply:
Throwing exceptions will not cause them to be emitted to the EventSource trace logging.  This seemed like a useful place to be explicit about it.

Reply:
AFAIK all CLR exceptions emit events already. At least ETW traces had them. Can we leverage that?

Reply:
Okay, since it isn't being logged already this does seem reasonable.